apps/website/src/components/toc/toc.tsx

```diff
@@ -1,33 +1,138 @@
+import { ContentHeading } from '@builder.io/qwik-city';
 import {
-  $,
-  QwikIntrinsicElements,
   component$,
   useSignal,
+  type JSXOutput,
   useVisibleTask$,
+  $,
 } from '@builder.io/qwik';
-import { ContentHeading } from '@builder.io/qwik-city';
-import { cn } from '@qwik-ui/utils';
 
 export const DashboardTableOfContents = component$(
   ({ headings }: { headings: ContentHeading[] }) => {
-    const itemIds = headings.map((item) => item.id);
-    const activeHeading = useActiveItem(itemIds);
-
     if (headings.length === 0) {
       return null;
     }
+    console.log(headings);
 
     return (
       <div class="space-y-2">
         <div class="font-medium">On This Page</div>
-        <Tree headings={headings} activeItem={activeHeading.value} />
+        <ul>
+          <TableOfContent headings={headings} />
+        </ul>
       </div>
     );
   },
 );
 
+import { cn } from '@qwik-ui/utils';
+type TableOfContentProps = { headings: ContentHeading[] };
+interface Node extends ContentHeading {
+  level: number;
+  children: Array<Node>;
+  activeItem: string;
+}
+type Tree = Array<Node>;
+export const TableOfContent = component$<TableOfContentProps>((props) => {
+  const infiniteStopper = JSON.parse(JSON.stringify(props.headings));
```

Comment:
I know I'm being picky, but would it be possible to get rid of this `JSON.parse(JSON.stringify(props.headings))` trick? I'm a huge fan of KISS and I'm a bit afraid this is a bit over-engineering an internal component that will be hard to debug in the future. I mean I'm not 100% against merging this but if there's a bug in the future, we might have to bring back the old TOC cause it's way more simple ^^

Maybe doing it more like shadcn? I thought what I did was exactly shadcn but you're right I missed something 😅 .. But maybe if we limit the logic to three levels of nesting then we don't need that trick somehow?

Reply:
You cannot modify any of the heading objects. If you do you cause  an infinite loop, which is really fun to debug when you're using recursive code. I could just write a loop that maps over the objects instead. Do you think that would be an improvement? 

> I'm a huge fan of KISS and I'm a bit afraid this is a bit over-engineering an internal component that will be hard to debug in the future

I don't think a looping non-recursive alternative would be any easier to debug. I can do it if that's what it would take to have your blessing though.  

> Maybe doing it more like shadcn?

I'm 99% sure we are doing it like shadcn. In fact, I think we could fork their recursive function and it would just work.

Reply:
If you can reproduce the shadcn version of it 100% without spending a week on it, this would be much more readable no? Can you give it a try?

> I could just write a loop that maps over the objects instead. Do you think that would be an improvement?
I'm not sure I understand the difference, but sounds simpler than the current impl 🙈

Reply:
> If you can reproduce the shadcn version of it 100% without spending a week on it, this would be much more readable no? Can you give it a try?

The shadcn uses a lot of ternaries. 
```js
return tree?.items?.length && level < 3 ?
```

```js
      {item.items?.length ? (
              <Tree tree={item} level={level + 1} activeItem={activeItem} />
            ) : null}
          </li>
```
I'm personally not a fan, but I need this part merged before going any further with the TOC so i'll do it.

Reply:
Thank you 🙏

Reply:
Should all be done. If it all looks good, I'll rebase main so we can merge.

Reply:
LGTM!